hail/src/main/scala/is/hail/rvd/RVDPartitioner.scala

```diff
@@ -13,13 +13,13 @@ class RVDPartitioner(
   // rangeBounds: Array[Interval[kType]]
   // rangeBounds is interval containing all keys within a partition
   val rangeBounds: Array[Interval],
-  allowedOverlap: Int
+  val allowedOverlap: Int
```

Comment:
`allowedOverlap` was intentionally not a field of the partitioner class. It represents a property that the `rangeBounds` may or may not have, and by passing it as an argument to the constructor, the caller can ask the constructor to enforce that the passed `rangeBounds` satisfy that property.

But I never saw any use cases for saving the `allowedOverlap` argument. You can always check if a partitioner satisfies a certain `allowedOverlap` with the `satisfiesAllowedOverlap` method. I don't understand what the `allowedOverlap` check is doing in `TableMultiWayZipJoin`, so maybe you did find a use case. Could you explain what you're trying to do?

Reply:
>  For that to work right, I need allowedOverlap == 0 (or to verify the partitions are in fact disjoint).

Ah, based on this, I think `rvd.partitioner.satisfiedAllowedOverlap(0)` does what you want.

Reply:
> Oh, allowedOverlap isn't even used, except in the assertion. Should we even keep it in?

Right, it's just asking for a stronger precondition to be checked on the range bounds during construction. Personally, I think it's worth keeping for that purpose.